mobify/platform-scaffold
# WEBDATA-241 Fetch shipping methods on form completion

For Hybris connector, there is a constraint that the API can only return shipping methods AFTER user submits shipping address information. Thus we need to find a way to delay `fetchShippingMethodsEstimate` for Hybris. 

In design meeting with @mitchquik, we decided to delay fetching shipping methods until user completes the shipping address form.

This PR introduce the ability to fetch shipping methods on form completion.

 **JIRA**: https://mobify.atlassian.net/browse/WEBDATA-241
 **Linked PRs**: ~~(links to corresponding PRs, optional)~~

## Changes
- Add placeholder to `ShippingMethod` component
- Fetch shipping methods on Shipping Address form field change

## Todos

## How to test-drive this PR
- Run `npm start`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- In `/app/connectors/_merlins-connector/checkout/commands.js` comment out line 67: `dispatch(receiveShippingMethods(shippingMethods))` (This line will be removed for Hybris connector), and don't forget to save :)
- Add stuff to cart, go to checkout 
- Verify there is a placeholder for shipping methods
- Verify the shipping methods are fetched and displayed after you filled out the shipping address form fields
